primary/src/grpc_server/configuration.rs

```diff
@@ -16,6 +16,45 @@ impl NarwhalConfiguration {
 
 #[tonic::async_trait]
 impl Configuration for NarwhalConfiguration {
+    async fn new_epoch(
+        &self,
+        request: Request<NewEpochRequest>,
+    ) -> Result<Response<Empty>, Status> {
+        let new_epoch_request = request.into_inner();
+        let epoch_number = new_epoch_request.epoch_number;
+        let validators = new_epoch_request.validators;
+        let mut parsed_input = vec![];
+        for validator in validators.iter() {
+            let proto_key = validator
+                .public_key
+                .as_ref()
+                .ok_or_else(|| Status::invalid_argument("Missing public key"))?;
+            let public_key: Ed25519PublicKey =
+                Ed25519PublicKey::from_bytes(proto_key.bytes.as_ref()).map_err(|err| {
+                    Status::invalid_argument(format!("Could not serialize: {:?}", err))
+                })?;
```

Comment:
maybe you could introduce a method like this https://github.com/MystenLabs/narwhal/blob/main/primary/src/grpc_server/proposer.rs#L30 so you can reuse it on the `new_network_info` as well?